create.ps1

```diff
@@ -139,267 +135,58 @@ function Get-ErrorMessage {
-                $exportData | Add-Member -MemberType NoteProperty -Name $aRefProperty.Name -Value $aRefProperty.Value -Force
-            }
-    
-            break
-        }
-        'Correlate' {
-            Write-Verbose "Correlating to AFAS employee [$($currentAccount.Medewerker)]"
-
-            if (-not($dryRun -eq $true)) {
-                # Set aRef object for use in futher actions
-                $aRef = [PSCustomObject]@{
-                    Medewerker     = $currentAccount.Medewerker
-                    Persoonsnummer = $currentAccount.Persoonsnummer
-                }
+    Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
 
-                $auditLogs.Add([PSCustomObject]@{
-                        # Action  = "" # Optional
-                        Message = "Successfully correlated to AFAS employee [$($currentAccount.Medewerker)]"
-                        IsError = $false
-                    })
-            }
-            else {
-                Write-Warning "DryRun: Would correlate to AFAS employee [$($currentAccount.Medewerker)]"
-            }
+    $auditLogs.Add([PSCustomObject]@{
+            # Action  = "" # Optional
+            Message = "Error querying AFAS employee where [$($correlationProperty)] = [$($correlationValue)]. Error Message: $($errorMessage.AuditErrorMessage)"
+            IsError = $true
+        })
 
-            # Define ExportData with account fields and correlation property 
-            $exportData = $account.PsObject.Copy() | Select-Object $storeAccountFields
-            # Add correlation property to exportdata
-            $exportData | Add-Member -MemberType NoteProperty -Name $correlationProperty -Value $correlationValue -Force
-            # Add aRef properties to exportdata
-            foreach ($aRefProperty in $aRef.PSObject.Properties) {
-                $exportData | Add-Member -MemberType NoteProperty -Name $aRefProperty.Name -Value $aRefProperty.Value -Force
-            }
-            
-            break
-        }
-    }
-}
-finally {
-    # Check if auditLogs contains errors, if no errors are found, set success to true
-    if (-NOT($auditLogs.IsError -contains $true)) {
-        $success = $true
-    }
-    
-    # Send results
-    $result = [PSCustomObject]@{
-        Success          = $success
-        AccountReference = $aRef
-        AuditLogs        = $auditLogs
-        PreviousAccount  = $previousAccount
-        Account          = $account
-    
-        # Optionally return data for use in other systems
-        ExportData       = $exportData
-    }
-    
-    Write-Output ($result | ConvertTo-Json -Depth 10)  
+    # Skip further actions, as this is a critical error
+    continue
 }
+
+# Check if auditLogs contains errors, if no errors are found, set success to true
+if (-NOT($outputContext.AuditLogs.IsError -contains $true)) {
```

Comment:
because at the start of the script $outputContext.Success = $true is set, even if the auditlogs contains IsError $true, then the $outputContext.Success is not set to $false?

Reply:
Copy error. resolved in new commit